### PR TITLE
feat: migrate buy now

### PR DESCRIPTION
**Pull request by domw30**
[Associated Notion Card](https://www.notion.so/zerotech/4985d8ec0c2f4eb4860678eec483e6f6?v=7fd94d39937343aa983ac94b7d7034d5&p=de4e3ce94afa471c8896a1b71145ee8b&pm=s)

-----

## 1. Pull request checklist

- [x] Notion card has been moved to the Code Review column
- [x] Notion card has a link to this PR
- [x] A reviewer has been assigned to the Notion card

## 2. PR type
- feature

## 3. What is the old behaviour?
Buy now is not currently available within the zOS container.

## 4. What is the new behaviour?
Buy now flow migrated and available to use 

Screenshot 2022-12-19 at 16 42 45

Screenshot 2022-12-19 at 16 43 11

Screenshot 2022-12-19 at 16 50 22

Screenshot 2022-12-19 at 17 41 22

### CODE REVIEW

File: src/lib/hooks/useZAuctionCheckTokensByDomain.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { useQuery } from 'react-query';
+
+import { useZnsSdk } from './useZnsSdk';
+import { BigNumber } from 'ethers';
+
+export const useZAuctionCheckTokensByDomain = (
```
[the remark this answers is not in the document]
Comment (domw30) [reply src/lib/hooks/useZAuctionCheckTokensByDomain.ts:6]: Yeah this was difficult to try and name. There are a number of zAuction Approval checks for skd.zauction, as shown below - this is why I moved away from `zAuction Approval`, if that makes sense?
Screenshot 2023-01-19 at 14 19 09

File: src/features/buy-now/BuyNowForm/FormSteps/FormSteps.module.scss [hunk 2]
```diff
@@ -0,0 +1,13 @@
+@import '../../../../../node_modules/@zero-tech/zui/styles/_colors.scss';
+@import '../../../../../node_modules/@zero-tech/zui/styles/_typography.scss';
```
[the remark this answers is not in the document]
Comment (domw30) [reply src/features/buy-now/BuyNowForm/FormSteps/FormSteps.module.scss:2]: Yeah for sure! Absolute path imports were proving a difficulty for us a couple of months back - this is how Brett left it at that time, then we had higher priority work to focus on. Happy to hear any suggestions on improving these - possibly for another task ? These paths are consistent at the moment throughout the zApps

File: src/features/buy-now/BuyNowForm/FormSteps/hooks/useFormSteps.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
+import { ReactNode } from 'react';
+
+import { Complete, Details } from '..';
+import { ApproveZAuction } from '../../../../ui';
+import { Wizard } from '@zero-tech/zui/components';
+
+export enum Step {
```
[the remark this answers is not in the document]
Comment (domw30) [reply src/features/buy-now/BuyNowForm/FormSteps/hooks/useFormSteps.tsx:7]: Agreed, I think these could be improved. This hook is used in each of the flows at the moment so I think we should leave as is for now - but this is something I'll return to once we've depreciated WWMM. Each of the flows can be refactored - they each share similar code.

File: src/features/buy-now/useBuyNowData.ts [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+	const { data: tokenBalance, isLoading: isLoadingTokenBalance } =
+		useUserTokenBalance(account, paymentToken?.id);
+
+	const { data: buyNowListingData, isLoading: isLoadingBuyNowListing } =
+		useBuyNowListing(domainId);
+
+	const title = metadata?.title;
+	const creator = domain?.minter;
+	const paymentTokenSymbol = paymentToken?.symbol ?? '';
+	const paymentTokenId = paymentToken?.id ?? '';
```
[the remark this answers is not in the document]
Comment (domw30) [reply src/features/buy-now/useBuyNowData.ts:40]: Yeah I like the idea of that. Something I'll include in refactoring when we get there - again, many of the flows use very similar hooks for data so I think at some point they can be merged into one hook and use object de-structuring to reduce the variable noise

File: src/lib/hooks/useZAuctionCheckTokensByDomain.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (padala) [reply src/lib/hooks/useZAuctionCheckTokensByDomain.ts:6]: Makes sense but should be okay to still name them. I’m okay with current name though

File: src/features/buy-now/BuyNowForm/FormSteps/FormSteps.module.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (padala) [reply src/features/buy-now/BuyNowForm/FormSteps/FormSteps.module.scss:2]: Got it. I’ll make a note of that will try to resolve it once I get some time

File: src/features/buy-now/BuyNowForm/FormSteps/hooks/useFormSteps.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (padala) [reply src/features/buy-now/BuyNowForm/FormSteps/hooks/useFormSteps.tsx:7]: Probably we can add a TODO comment.

File: src/features/buy-now/useBuyNowData.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (padala) [reply src/features/buy-now/useBuyNowData.ts:40]: Makes sense

**Review by padala — APPROVED** [review]

**Pull request merged**